apps/engine/lib/engine/release_tasks/contract.ex

```diff
@@ -31,33 +32,42 @@ defmodule Engine.ReleaseTasks.Contract do
     default_url = config |> Keyword.fetch!(:ethereumex) |> Keyword.fetch!(:url)
     rpc_url = "ETHEREUM_RPC_URL" |> get_env() |> Validators.url("ETHEREUM_RPC_URL", default_url)
 
-    [
-      payment_exit_game,
-      eth_vault,
-      erc20_vault,
-      min_exit_period_seconds,
-      contract_semver,
-      child_block_interval,
-      contract_deployment_height
-    ] = external_data(plasma_framework, tx_hash, rpc_url)
-
-    Config.Reader.merge(config,
-      engine: [
-        rpc_url: rpc_url,
-        authority_address: authority_address,
-        plasma_framework: plasma_framework,
-        eth_vault: eth_vault,
-        erc20_vault: erc20_vault,
-        payment_exit_game: payment_exit_game,
-        min_exit_period_seconds: min_exit_period_seconds,
-        contract_semver: contract_semver,
-        child_block_interval: child_block_interval,
-        contract_deployment_height: contract_deployment_height
-      ]
-    )
+    contracts_data = get_external_data(plasma_framework, tx_hash, rpc_url)
```

Comment:
this seems like it doesn't really tell what it's really doing, you're merging two funcitonalities into the same function -> reading from database and deciding if it needs to fetch the data externally. perhaps you should rethink this.

also, your approach does not obey the stacking arhitecture, you need to think about not having postgres on boot, similarly to how I handled not having geth/infura on boot:
```
defp call(plasma_framework, signature, args, opts) do
    case Rpc.call_contract(plasma_framework, signature, args, opts) do
      {:ok, _data} = result ->
        result

      reason ->
        # if ethereum client isn't ready yet, we wait until it comes back
        {timeout, backoff} = Backoff.backoff(Process.get(:backoff))
        %Backoff{} = Process.put(:backoff, backoff)
        rpc_url = Keyword.get(opts, :url)

        _ =
          Logger.error(
            "Ethereum client not available #{rpc_url}. Getting #{inspect(reason)}. Will wait for #{timeout} ms."
          )

        :ok = Process.sleep(timeout)
        call(plasma_framework, signature, args, opts)
    end
  end
```

Reply:
@InoMurko why do want to handle not having postgres on boot? Infra won't restart the container if app fails to start on a release task?

Reply:
Because we’re following 
https://medium.com/@jlouis666/stacking-theory-for-systems-design-2450e6300689

Reply:
If you don't want to read the essay, tho I think you should, the idea is to have a reactive system on all layers. The layer you're working on right now allows us to decouple service boot ordering. Without necessarily be bound by the deployment mechanism.

Reply:
I read it, thanks.